pkg/controller/scyllacluster/resource.go

```diff
@@ -552,14 +552,14 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, existing
 								},
 							},
 							// Before a Scylla Pod is stopped, execute nodetool drain to
-							// flush the memtable to disk and stop listening for connections.
-							// This is necessary to ensure we don't lose any data and we don't
-							// need to replay the commitlog in the next startup.
+							// flush the memtable to disk, finish existing requests and stop listening for connections.
+							// Sleep is required to give chance to Load Balancers to acknowledge Pod going down with their
+							// probes.
 							Lifecycle: &corev1.Lifecycle{
 								PreStop: &corev1.LifecycleHandler{
 									Exec: &corev1.ExecAction{
 										Command: []string{
-											"/bin/sh", "-c", "PID=$(pgrep -x scylla);supervisorctl stop scylla; while kill -0 $PID; do sleep 1; done;",
```

Comment:
Can you clarify why this was needed before and not after? Does the SIGTERM (after preStop hook) propagate correctly now?

Reply:
synced on a call, this is fine as SIGTERM now works